pkg/registry/limitrange/rest.go

```diff
@@ -17,137 +17,61 @@ limitations under the License.
-		return nil, false, fmt.Errorf("invalid object type")
-	}
-
-	if !api.ValidNamespace(ctx, &limitRange.ObjectMeta) {
-		return nil, false, errors.NewConflict("limitRange", limitRange.Namespace, fmt.Errorf("LimitRange.Namespace does not match the provided context"))
-	}
-
-	oldObj, err := rs.registry.Get(ctx, limitRange.Name)
-	if err != nil {
-		return nil, false, err
-	}
-
-	editLimitRange := oldObj.(*api.LimitRange)
-
-	// set the editable fields on the existing object
-	editLimitRange.Labels = limitRange.Labels
-	editLimitRange.ResourceVersion = limitRange.ResourceVersion
-	editLimitRange.Annotations = limitRange.Annotations
-	editLimitRange.Spec = limitRange.Spec
-
-	if errs := validation.ValidateLimitRange(editLimitRange); len(errs) > 0 {
-		return nil, false, errors.NewInvalid("limitRange", editLimitRange.Name, errs)
-	}
-
-	if err := rs.registry.UpdateWithName(ctx, editLimitRange.Name, editLimitRange); err != nil {
-		return nil, false, err
-	}
-	out, err := rs.registry.Get(ctx, editLimitRange.Name)
-	return out, false, err
 }
 
-// Delete deletes the LimitRange with the specified name
-func (rs *REST) Delete(ctx api.Context, name string) (runtime.Object, error) {
-	obj, err := rs.registry.Get(ctx, name)
-	if err != nil {
-		return nil, err
-	}
-	_, ok := obj.(*api.LimitRange)
-	if !ok {
-		return nil, fmt.Errorf("invalid object type")
-	}
-	return rs.registry.Delete(ctx, name, nil)
+func (limitrangeStrategy) PrepareForUpdate(obj, old runtime.Object) {
 }
 
-// Get gets a LimitRange with the specified name
-func (rs *REST) Get(ctx api.Context, name string) (runtime.Object, error) {
-	obj, err := rs.registry.Get(ctx, name)
-	if err != nil {
-		return nil, err
-	}
-	limitRange, ok := obj.(*api.LimitRange)
-	if !ok {
-		return nil, fmt.Errorf("invalid object type")
-	}
-	return limitRange, err
+func (limitrangeStrategy) Validate(ctx api.Context, obj runtime.Object) fielderrors.ValidationErrorList {
+	limitRange := obj.(*api.LimitRange)
+	return validation.ValidateLimitRange(limitRange)
 }
 
-func (rs *REST) getAttrs(obj runtime.Object) (objLabels labels.Set, objFields fields.Set, err error) {
-	return labels.Set{}, fields.Set{}, nil
+func (limitrangeStrategy) AllowCreateOnUpdate() bool {
+	return true
 }
 
-func (rs *REST) List(ctx api.Context, label labels.Selector, field fields.Selector) (runtime.Object, error) {
-	return rs.registry.ListPredicate(ctx, &generic.SelectionPredicate{label, field, rs.getAttrs})
+func (limitrangeStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object) fielderrors.ValidationErrorList {
+	limitRange := obj.(*api.LimitRange)
+	return validation.ValidateLimitRange(limitRange)
 }
 
-func (rs *REST) Watch(ctx api.Context, label labels.Selector, field fields.Selector, resourceVersion string) (watch.Interface, error) {
-	return rs.registry.WatchPredicate(ctx, &generic.SelectionPredicate{label, field, rs.getAttrs}, resourceVersion)
+func (limitrangeStrategy) AllowUnconditionalUpdate() bool {
+	return false
```

Comment:
I think its fine to set this to true
As far as i know we support unconditional update for all resources.